tools/modules/framework/module_data_files.sh

```
@@ -0,0 +1,390 @@
+	["generate_data_files,port"]=""
+	["generate_data_files,arch"]=""
+)
+# Use production array to verify a parent/group and subgroup/subsubgroup keys are valid
+# may be used to verify other keys if needed
+function generate_data_files() {
+	local generator=$1
+	local i=0
+	local g=0
+
+	features=()
+	for key in "${!module_options[@]}"; do
```

Shellcheck (0.10.0)

[warning] module_options is referenced but not assigned.

(SC2154)